src/temp_sensor.rs

```diff
@@ -0,0 +1,149 @@
+            }
+        }
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl From<temperature_sensor_clk_src_t> for TemperatureSensorClockSource {
+    fn from(value: temperature_sensor_clk_src_t) -> Self {
+        match value {
+            #[allow(non_upper_case_globals)]
+            soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL => Self::XTAL,
+            #[allow(non_upper_case_globals)]
+            soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST => {
+                Self::RcFast
+            }
+            // TODO: Perhaps the default value should be mapped explicitly
+            // and all other (u32) values should cause a failure
+            _ => Self::Default,
+        }
+    }
+}
+
+// -- TemperatureSensorConfig --
+
+#[cfg(not(any(esp32)))]
+#[derive(Copy, Clone)]
+/// Rust wrapper for `temperature_sensor_config_t`
+pub struct TemperatureSensorConfig {
+    // TODO: check int size
+    pub range_min: i32,
+    pub range_max: i32,
+    pub clk_src: TemperatureSensorClockSource,
+}
+
+#[cfg(not(any(esp32)))]
+impl From<temperature_sensor_config_t> for TemperatureSensorConfig {
+    fn from(value: temperature_sensor_config_t) -> Self {
+        TemperatureSensorConfig {
+            range_min: value.range_min,
+            range_max: value.range_max,
+            clk_src: value.clk_src.into(),
+        }
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl Into<temperature_sensor_config_t> for TemperatureSensorConfig {
+    fn into(self) -> temperature_sensor_config_t {
+        temperature_sensor_config_t {
+            clk_src: self.clk_src.into(),
+            range_max: self.range_max,
+            range_min: self.range_min,
+        }
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl Default for TemperatureSensorConfig {
+    fn default() -> Self {
+        TemperatureSensorConfig {
+            range_min: -10,
+            range_max: 80,
+            clk_src: TemperatureSensorClockSource::Default,
+        }
+    }
+}
+
+// -- TemperatureSensorDriver --
+
+#[cfg(not(any(esp32)))]
+pub struct TemperatureSensorDriver {
+    ptr: temperature_sensor_handle_t,
+}
+
+#[cfg(not(any(esp32)))]
+impl TemperatureSensorDriver {
+    pub fn new(config: TemperatureSensorConfig) -> Result<Self, EspError> {
+        let mut sensor = core::ptr::null_mut();
+        esp!(unsafe { temperature_sensor_install(&config.into(), &mut sensor) })?;
+        Ok(TemperatureSensorDriver { ptr: sensor })
+    }
+
+    pub fn enable(&mut self) -> Result<(), EspError> {
+        esp!(unsafe { temperature_sensor_enable(self.ptr) })
+    }
+
+    pub fn disable(&mut self) -> Result<(), EspError> {
+        esp!(unsafe { temperature_sensor_disable(self.ptr) })
+    }
+
+    pub fn get_celsius(&self) -> Result<f32, EspError> {
+        let mut val = 0.0;
+        esp!(unsafe { temperature_sensor_get_celsius(self.ptr, &mut val) })?;
+        Ok(val)
+    }
+
+    pub fn get_fahrenheit(&self) -> Result<f32, EspError> {
```

Comment:
Do we really need this?

Reply:
Personally I would not, but I just did the rebase so I'd wait for the original author. @GamerBene19 do you have time to complete it?

Reply:
I don't follow. :) 
Do you plan to finish this PR, or you just rebased and plan to leave it as-is, leaving it in open state? If the latter, it would likely just rot as the original PR did, as nobody else might have the time to complete it for you.

Reply:
> Do we really need this?

(Also replying for kelvin below)
Since I "passed through" the naming of the implementation from espressif (which contains "_celcius") I thought it wouldn't hurt to have these functions available to users of the library for convenience.
If that's not a thing you do in this project we can leave it out. I don't have any strong preference in either direction.

> do you have time to complete it?

See https://github.com/esp-rs/esp-idf-hal/pull/337#issuecomment-2336697296: TL;DR Yes, but I need some assistance.

Reply:
Sure and I'll assist. As I posted in the other comment - it does not need a huge amount of work anyway.